plugins/statistics.koplugin/main.lua

```diff
@@ -1,32 +1,41 @@
-local Widget = require("ui/widget/widget")
-local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local DocSettings = require("docsettings")
+local Event = require("ui/event")
+local InfoMessage = require("ui/widget/infomessage")
 local KeyValuePage = require("ui/widget/keyvaluepage")
-local UIManager = require("ui/uimanager")
-local Screen = require("device").screen
+local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ReaderProgress = require("readerprogress")
+local ReadHistory = require("readhistory")
+local SQ3 = require("lua-ljsqlite3/init")
 local TimeVal = require("ui/timeval")
-local DataStorage = require("datastorage")
+local UIManager = require("ui/uimanager")
+local Widget = require("ui/widget/widget")
 local lfs = require("libs/libkoreader-lfs")
-local DEBUG = require("dbg")
-local T = require("ffi/util").template
-local joinPath = require("ffi/util").joinPath
-local _ = require("gettext")
+local logger = require("logger")
 local util = require("util")
-local tableutil = require("tableutil")
-local ReadHistory = require("readhistory")
-local DocSettings = require("docsettings")
-local ReaderProgress = require("readerprogress")
-local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local _ = require("gettext")
+local joinPath = require("ffi/util").joinPath
+local Screen = require("device").screen
+local T = require("ffi/util").template
 
--- a copy of page_max_read_sec
-local page_max_time
+local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local db_location = DataStorage:getSettingsDir() .. "/book.sqlite3"
 
 local ReaderStatistics = Widget:extend{
-    last_time = nil,
-    page_min_read_sec = 5,
-    page_max_read_sec = 90,
-    current_period = 0,
-    pages_current_period = 0,
+    page_min_read_sec = 1,
+    page_max_read_sec = 180,
```

Comment:
Well, I have two things on this:

First, shouldn't this new statistics use old user settings for statistics? We can define (or we could before) our timings.

Second, time to read page depends very much on many things. I personally want smallest font I can see (with time, it surely increase in size) and if you have bigger reader (i.e. bigger screen), that can take some time. For me, at the moment, my average page takes about 3 minutes.

In this case, both values are "wrong" for me. My minimum is 15, maximum is 900. I think it would be best if timings are user customizable.